api/v1alpha1/siteconfig_types.go

```diff
@@ -0,0 +1,461 @@
+	// +optional
+	// +kubebuilder:default:=disabled
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode bmh_v1alpha1.AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
+
+	// RootDeviceHints specifies the device for deployment.
+	// Identifiers that are stable across reboots are recommended, for example, wwn: <disk_wwn> or deviceName: /dev/disk/by-path/<device_path>
+	// +optional
+	RootDeviceHints bmh_v1alpha1.RootDeviceHints `json:"rootDeviceHints,omitempty"`
+
+	// NodeNetwork is a set of configurations pertaining to the network settings for the node.
+	// +optional
+	NodeNetwork aiv1beta1.NMStateConfigSpec `json:"nodeNetwork,omitempty"`
+
+	// NodeLabels allows the specification of custom roles for your nodes in your managed clusters.
+	// These are additional roles are not used by any OpenShift Container Platform components, only by the user.
+	// When you add a custom role, it can be associated with a custom machine config pool that references a specific configuration for that role.
+	// Adding custom labels or roles during installation makes the deployment process more effective and prevents the need for additional reboots
+	// after the installation is complete.
+	// +optional
+	NodeLabels map[string]string `json:"nodeLabels,omitempty"`
+
+	// Hostname is the desired hostname for the host
+	// +required
+	HostName string `json:"hostName,omitempty"`
+
+	//TODO: remove NodeIp
+	NodeIp string `json:"nodeIp,omitempty"`
+
+	// Provide guidance about how to choose the device for the image being provisioned.
+	// +kubebuilder:default:=UEFI
+	BootMode bmh_v1alpha1.BootMode `json:"bootMode,omitempty"`
+
+	// UserData holds the reference to the Secret containing the user data to be passed to the host before it boots.
+	// +optional
+	UserData corev1.SecretReference `json:"userData,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's coreos installer args
+	// +optional
+	InstallerArgs string `json:"installerArgs,omitempty"`
+
+	// Json formatted string containing the user overrides for the host's ignition config
+	// IgnitionConfigOverride enables the assignment of partitions for persistent storage.
+	// Adjust disk ID and size to the specific hardware.
+	// +optional
+	IgnitionConfigOverride string `json:"ignitionConfigOverride,omitempty"`
+
+	// +kubebuilder:default:=master
+	Role string `json:"role,omitempty"`
+
+	// +optional
+	CrAnnotations CrAnnotations `json:"crAnnotations,omitempty"`
+
+	// +optional
+	DiskPartition []DiskPartition `json:"diskPartition,omitempty"`
+
+	// IronicInspect is used to specify if automatic introspection carried out during registration of BMH is enabled or disabled
+	// +kubebuilder:default:=""
+	// +optional
+	IronicInspect IronicInspect `json:"ironicInspect,omitempty"`
+
+	// +required
+	TemplateRefs []TemplateRef `json:"templateRefs,omitempty"`
+}
+
+const (
+	SNO      = "sno"
```

Comment:
If these are the cluster type options, you can create a separate typedef as a string and create these as `CluterType`s.

Either way I'd prefix these like `ClusterTypeSNO` and `ClusterTypeStandard`.

Reply:
Changed!